src/index.js

```diff
@@ -46,27 +61,115 @@ const handleItemMenuClick = () => {
       input.focus();
       parent.classList.add('active');
 
+      input.addEventListener('change', (e) => {
+        e.preventDefault();
+        let fromLocal = JSON.parse(window.localStorage.getItem('todo-tasks'));
+
+        fromLocal = fromLocal.map((task) => {
+          if (task.id === checkBoxId) {
+            const obj = {
+              ...task,
+              description: input.value,
+              completed: false,
+            };
+            return obj;
+          }
+          return task;
+        });
+        // we can as well rerender the updated item to the ui
+        document.querySelector('.todo-list').innerHTML = '';
+        populateList(fromLocal);
+        handleOnCheckboxClick();
+        handleItemMenuClick();
+
+        window.localStorage.setItem('todo-tasks', JSON.stringify(fromLocal));
+      });
+
       deleteIcon.addEventListener('click', () => {
         parent.parentElement.removeChild(parent);
+        // we also update the local storage accordingly
+        const refId = parent.children[1].id;
+
+        let fromLocalStorage = window.localStorage.getItem('todo-tasks');
+        if (fromLocalStorage.length) {
+          fromLocalStorage = JSON.parse(fromLocalStorage);
+
+          fromLocalStorage = fromLocalStorage.filter((task) => {
+            const condition = task.id !== Number(refId);
+            return condition;
+          });
+
+          // reset the ids of the remaining tasks
+          fromLocalStorage = fromLocalStorage.map((task, index) => {
+            const result = {
+              ...task,
+              id: index + 1,
+            };
+            return result;
+          });
+
+          window.localStorage.setItem(
+            'todo-tasks',
+            JSON.stringify(fromLocalStorage),
+          );
+        }
```

Comment:
[OPTIONAL] I recommend adding the logic of removing a task to a function called  `removeTask()` and calling this function in the event listener so that your code is clear and more organized.